monitor/monitor.go

```diff
@@ -241,49 +235,132 @@ func (m *Monitor) monitorDPoSIntendWithdraw() {
 	}
 }
 
-func (m *Monitor) monitorCelerLedgerIntendWithdraw() {
+func (m *Monitor) monitorCelerLedgerIntendSettle() {
 	_, err := m.ethMonitor.Monitor(
 		&monitor.Config{
-			EventName:     string(IntendWithdraw),
+			EventName:     string(IntendSettle),
 			Contract:      m.ledgerContract,
-			StartBlock:    m.ethMonitor.GetCurrentBlockNumber(),
-			CheckInterval: eventCheckInterval(IntendWithdrawChannel),
+			StartBlock:    m.getCurrentBlockNumber(),
+			CheckInterval: eventCheckInterval(IntendSettle),
 		},
 		func(cb monitor.CallbackID, eLog ethtypes.Log) {
-			log.Infof("Catch event IntendWithdrawChannel, tx hash: %x", eLog.TxHash)
-			event := NewEvent(IntendWithdrawChannel, eLog)
-			dberr := m.dbSet(GetPullerKey(eLog), event.MustMarshal())
-			if dberr != nil {
-				log.Errorln("db Set err", dberr)
+			log.Infof("Catch event IntendSettle, tx hash: %x", eLog.TxHash)
+			err := m.dbSet(GetPullerKey(eLog.TxHash), NewEvent(IntendSettle, eLog).MustMarshal())
+			if err != nil {
+				log.Errorln("db Set err", err)
 			}
-			dberr = m.dbSet(GetGuardKey(eLog), event.MustMarshal())
-			if dberr != nil {
-				log.Errorln("db Set err", dberr)
+			e, err := m.ethClient.Ledger.ParseIntendSettle(eLog)
+			if err != nil {
+				log.Errorln("ParseIntendSettle err", err)
+				return
+			}
+			addresses, seqNums, err := m.ethClient.Ledger.GetStateSeqNumMap(&bind.CallOpts{}, e.ChannelId)
+			if err != nil {
+				log.Errorf("Query StateSeqNumMap for cid %x err: %s", e.ChannelId, err)
+				return
+			}
+			for i, simplexReceiver := range addresses {
+				key := GetGuardKey(e.ChannelId, simplexReceiver)
```

Comment:
Can we put the loop content into a function

Reply:
done